ofrak_core/ofrak/core/esp/app.py

```diff
@@ -0,0 +1,959 @@
+                    await resource.create_child_from_view(
+                        signature, data_range=Range.from_size(signature_offset, 64004).translate(esp_config.offset)
+                    )
+            # Process application description if present
+            if app_desc:
+                app_description, magic_word = self.__parse_app_description(app_desc)
+                if magic_word == 0xABCD5432:
+                    await resource.create_child_from_view(
+                        app_description,
+                        data_range=Range.from_size(
+                            ESP_APP_HEADER_SIZE + ESP_APP_EXTENDED_HEADER_SIZE + ESP_APP_SEGMENT_HEADER_SIZE,
+                            256,
+                        ).translate(esp_config.offset),
+                    )
+            elif bootloader_desc:
+                bootloader_description, magic_byte = self.__parse_bootloader_description(
+                    bootloader_desc
+                )
+                if magic_byte == 80:
+                    await resource.create_child_from_view(
+                        bootloader_description,
+                        data_range=Range.from_size(
+                            ESP_APP_HEADER_SIZE + ESP_APP_EXTENDED_HEADER_SIZE + ESP_APP_SEGMENT_HEADER_SIZE, 80
+                        ).translate(esp_config.offset),
+                    )
+        finally:
+            if hasattr(esp_config, "f"):
+                esp_config.f.close()
+
+    def __parse_flash_bits(self, image) -> Tuple[int, int]:
+        """
+        Extracts and returns flash size and frequency bits from the firmware image.
+
+        :param image: The firmware image object
+        :return: A tuple containing the flash size bits and flash frequency bits
+        """
+        flash_s_bits = image.flash_size_freq & 0xF0
+        flash_fr_bits = image.flash_size_freq & 0x0F
+        return flash_s_bits, flash_fr_bits
+
+    def __parse_app_description(self, app_desc: bytes) -> Tuple[ESPAppDescription, int]:
+        """
+        Parses and returns the application description from segment data.
+
+        :param app_desc: The application description data extracted from the DROM segment
+        :return: A tuple of (ESPAppDescription object, magic word)
+        """
+        APP_DESC_STRUCT_FMT = "<II8s32s32s16s16s32s32s80s"
+        unpacked_data = struct.unpack(APP_DESC_STRUCT_FMT, app_desc)
+        return (
+            ESPAppDescription(
+                magic=unpacked_data[0],
+                secure_version=unpacked_data[1],
+                reserv1=unpacked_data[2],
+                version=unpacked_data[3],
+                project_name=unpacked_data[4],
+                time=unpacked_data[5],
+                date=unpacked_data[6],
+                idf_ver=unpacked_data[7],
+                app_eld_sha256=unpacked_data[8],
+                reserv2=unpacked_data[9],
+            ),
+            unpacked_data[0],
+        )
```

Comment:
```suggestion
        return (
            ESPAppDescription(*unpacked_data),
            unpacked_data[0],
        )
```

Would this work?

Reply:
Same comment applies to other places where struct.unpack output is manually passed to dataclass constructors, including immediately below